### ISSUE TITLE
NullReferenceException in ReflectionAnalyzer

### ISSUE THREAD

**Issue by Ollhax** [microsoft/Microsoft.Unity.Analyzers#124]
#### Bug description

With the following code:

```
public class Test : MonoBehaviour
{
	private Collider[] colliders = new Collider[4];
	private void Update()
	{
		var go = colliders[0].gameObject;
	}
}
```

I get this warning in the error list in visual studio:

```
Warning	AD0001	Analyzer 'Microsoft.Unity.Analyzers.ReflectionAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
Exception occurred with following context:
Compilation: Assembly-CSharp
SyntaxTree: C:\Dev\Projects\Test\Assets\Scripts\Test.cs
SyntaxNode: private void Update() { var go = ... [MethodDeclarationSyntax]@[141..210) (7,2)-(10,3)

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Unity.Analyzers.ReflectionAnalyzer.AnalyzeMethodBody(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax method)
   at Microsoft.Unity.Analyzers.ReflectionAnalyzer.AnalyzeMethodDeclaration(SyntaxNodeAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__62`1.<ExecuteSyntaxNodeAction>b__62_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```
- Unity: 20.2.1f1
- VS: 16.8.3

**Comment by sailro** [comment]
Hi @Ollhax, I'm not able to repro. I even tried to retype every character instead of copy/pasting the whole code block in case this can be related to an unexpected code model while you type.

If you are still able to repro, could you send me the whole project ? Perhaps there is something else triggering the issue.

**Comment by sailro** [comment]
I was finally able to repro. Re-using your example and then using a quick fix to replace `go` with the discard `-` character. In this case the containing namespace is null in the codemodel which is unexpected...
